### PR TITLE
fix(correction): #1713 pre-judge redaction multi-dir + stale-applying recovery

### LINKED ISSUE #1713: Follow-up: pre-judge redaction filter should consult target namespace rules + stale-applying plan recovery
## Context

Two review threads on PR #1705 are deferred as marginal/defense-in-depth. Both are correct observations but the fixes are not blocking for the correction contract feature (#1669/#1672/#1678).

## Item 1: Pre-judge redaction filter consults only source namespace

**Thread:** chatgpt-codex-connector P1 + kilo-code-bot WARNING on PR #1705.

The pre-judge redaction filter at `orchestrator.ts:14870` loads `redactionRulesFor(storage.dir)` — only the source namespace. The persist gate at line 15174 correctly loads both `sourceStorageDir` and `targetStorage.dir`.

**Why deferred:** `collectJudgeTrainingPairs` is off by default. Judge-training JSONL lives in `~/.remnic/judge-training/` (local home dir, not committed/synced/bundled into memory exports). The persist gate already prevents the memory write. The pre-judge filter already catches source-namespace never_store rules (the common case). The cross-namespace target case requires resolving per-fact target storage dirs at the batch pre-filter point, duplicating routing logic.

**Fix:** At the pre-judge batch point, also load shared-namespace rules and/or resolve per-fact routed target dirs via `this.storageRouter.storageFor()` using `preRoutedCategories`.

## Item 2: Stale `applying` plan recovery on startup

**Thread:** chatgpt-codex-connector P1 on PR #1705.

When apply reaches the optimistic `applying` mark and then the process exits, the pending-plan file retains `redaction_rule.pattern` intact. `applying` plans are rejected on future apply attempts and not returned by `listPending`.

**Why deferred:** Scrubbing at `applying` would break the executor — it reloads the plan via `loadPlan` between `applying` and `applied` and needs the real pattern to call `registerRedactionRule`. The crash-during-apply scenario is rare, and the pending file lives in a local state dir (not committed/synced).

**Fix:** On startup, detect stale `applying` plans past a TTL and either retry or discard+scrub them.

### PR DESCRIPTION
**Pull request by joshuaswarren**
Closes #1713 — both deferred review-thread items from PR #1705.

## Item 1: Pre-judge redaction filter consults target namespace rules

The pre-judge redaction filter previously loaded rules from only the source namespace dir. A never_store pattern registered under a cross-namespace target was missed at the batch pre-filter point (only caught later at the persist gate).

**Fix:** The pre-judge filter now collects routed target namespaces during routing (preRoutedNamespaces) and resolves their storage dirs via storageRouter.storageFor(). Combined with the shared namespace dir, redactionRulesFor(...preJudgeDirs) now catches cross-namespace never_store patterns at the batch pre-filter point — matching the persist gate behavior.

## Item 2: Stale applying correction-plan recovery

When apply reaches the optimistic applying mark and then the process dies, the pending-plan file retains redaction_rule.pattern intact and the plan is stuck in applying.

**Fix:**
- CorrectionPlan gains an applyingAt ISO timestamp
- markConsumed stamps applyingAt on entering applying, clears it on terminal transitions
- CorrectionPlanner.recoverStaleApplyingPlans scans for plans stuck in applying past a 10-min TTL and discards+scrubs them. Pre-fix plans (no applyingAt) fall back to expiresAt
- CorrectionService wraps the multi-namespace sweep; orchestrator initialize() wires it in as best-effort startup maintenance (uses configuredNamespaceList() chokepoint)

## Chokepoints used / not applicable
- storageRouter.storageFor — namespace resolution
- configuredNamespaceList / getConfiguredNamespaces — namespace set derivation
- serializeMutations — markConsumed write path (existing, unchanged)
- No new config flags — scattered-config-flag-reads ratchet unchanged (536)

## Ratchet
- orchestrator.ts +33 LOC: both items are necessary bug fixes with no extracted module to delegate to. Baseline updated with justification.
- scatteredConfigFlagReads: unchanged (536)

## Tests
- 6 new planner tests: applyingAt stamp/clear, stale recovery (discard+scrub), fresh-left-alone, empty-namespace, pre-fix expiresAt fallback
- 2 new redaction tests: combined source+target rules catch cross-namespace patterns; source-only misses them (the bug)
- Entity-hardening: 246/246 pass
- Full correction suite: 94/94 pass
- Build (ESM+DTS) + check-types clean

---

> [!NOTE]
> **Medium Risk**
> Changes extraction pre-judge filtering and startup correction maintenance on stuck plans; both affect never-store/redaction handling but are targeted fixes with fail-open error handling.
> 
> **Overview**
> Fixes two follow-ups from #1713: **pre-judge redaction** now respects where each fact will land, and **stale `applying` correction plans** are recovered on startup.
> 
> **Pre-judge redaction (Item 1):** During `persistExtraction`, routing now records a per-fact target namespace. The extraction-judge candidate loop merges redaction rules from that namespace (via `storageRouter.storageFor`) with source rules, and mirrors write-time **scope=global → shared** routing so shared-only never-store patterns are filtered before the judge—not only at the persist gate.
> 
> **Stale applying recovery (Item 2):** `CorrectionPlan` gains optional `applyingAt`; `markConsumed` sets it on `applying` and clears it on terminal states. `recoverStaleApplyingPlans` discards+scrubs plans stuck in `applying` past a 10-minute TTL (or past `expiresAt` for older plans without `applyingAt`). `CorrectionService.recoverStaleApplyingPlans` sweeps namespaces without requiring the correction feature flag; orchestrator init runs this best-effort over configured + catalog namespaces.
> 
> Tests cover applyingAt lifecycle, recovery edge cases, combined redaction dirs, and scope-routing behavior; `orchestrator.ts` watchlist LOC ratchet is bumped.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 7cdfe9eeb18c69cfd83b88c76ba7960a4c7f0b33. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: packages/remnic-core/src/orchestrator.ts [hunk 1]
```diff
@@ -15069,10 +15109,25 @@ export class Orchestrator {
           ) {
             continue;
           }
-          if (preJudgeRedactionRules.length > 0) {
+          // #1713 P2 (cursor): per-fact rules — load this fact's routed target
+          // namespace rules FIRST, then check. This catches target-only rules
+          // even when the base (source+shared) rules are empty (threads acc58c42
+          // + PBJEe/PBKAj).
+          let factRedactionRules = preJudgeRedactionRules;
+          const factNs = preRoutedNamespaceByFact[fi];
```
[the remark this answers is not in the document]
Comment (joshuaswarren) [reply packages/remnic-core/src/orchestrator.ts]: Fixed in 72c0269ba. The pre-judge per-fact redaction block now mirrors the write loop's scope-routing decision: when no routing rule set an explicit namespace (`!factNs`), `extractionScopeClassification` is on, the fact is `scope === "global"`, `profileAllowsSharedWrites`, and the source namespace is not already shared, the per-fact target is set to the shared namespace so its redaction rules are loaded and consulted alongside the source (base) rules — exactly the conditions the write loop uses at orchestrator.ts scope-routing. A scope=global fact routed to shared with a never-store rule living only under shared is now withheld at pre-judge, never reaching the judge/training path.

To stay ratchet-neutral (ground rule 5: no new scattered `config.*Enabled` reads), `namespacesEnabled` is hoisted to a single const at the top of persistExtraction and shared by both the pre-judge mirror and the write-loop gate, so `scatteredConfigFlagReads` stays at 511.

Tests added in extraction-redaction-rules.test.ts: a scope=global fact resolves to the shared target and the combined source+shared rules catch a shared-only never-store (the gap); a scope-local fact is not scope-routed and consults no shared rules. Typecheck, build, correction (115), planner (21), redaction (21), entity-hardening (246), and check-ratchets all green.
